### PR TITLE
[Feat](Nereids) support json and jsonb datatype

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
## Proposed changes

Feature:
support jsonb and json type in nereids

Document:
this feature supports these two datatype in nereids optimizer like original planner, the sql reference is same as before
[JSON - Apache Doris](https://doris.apache.org/zh-CN/docs/dev/sql-manual/sql-reference/Data-Types/JSON)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CheckDataTypes.java [hunk 1]
```diff
@@ -95,8 +95,6 @@ private void checkTypes(DataType dataType) {
                 checkTypes(((MapType) dataType).getValueType());
             } else if (dataType instanceof StructType) {
                 ((StructType) dataType).getFields().forEach(f -> this.checkTypes(f.getDataType()));
-            } else if (UNSUPPORTED_TYPE.contains(dataType.getClass())) {
-                throw new AnalysisException(String.format("type %s is unsupported for Nereids", dataType));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CheckDataTypes.java:99]: should not remove this

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonbExtract.java [hunk 2]
```diff
@@ -39,24 +40,25 @@ public class JsonbExtract extends ScalarFunction
...
     }
 
     /**
      * withChildren.
      */
     @Override
     public JsonbExtract withChildren(List<Expression> children) {
-        Preconditions.checkArgument(children.size() == 2);
-        return new JsonbExtract(children.get(0), children.get(1));
+        Preconditions.checkArgument(children.size() >= 2);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/JsonbExtract.java:59]: add check msg

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ScalarFunctionVisitor.java [hunk 3]
```diff
@@ -1072,6 +1073,10 @@ default R visitJsonbParseNullableErrorToValue(JsonbParseNullableErrorToValue fun
         return visitScalarFunction(function, context);
     }
 
+    default R visitJsonbValid(JsonbValid jsonbValid, C context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/visitor/ScalarFunctionVisitor.java:1076]: add in lexicographical order

File: regression-test/data/nereids_p0/json_p0/test_json.csv [hunk 4]
```diff
@@ -0,0 +1,25 @@
+1	\N
+2	null
+3	true
```
Comment (morrySnow) [inline regression-test/data/nereids_p0/json_p0/test_json.csv:3]: add all jsonb function test in regression-test/suites/nereids_function_p0/scalar_function

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Pull request merged**